lib/galaxy/objectstore/__init__.py

```diff
@@ -1577,3 +1585,27 @@ def set_dataset_object_store_id(self, dataset, require_shareable=True):
         except ObjectInvalid:
             raise Exception("Unable to create output dataset: object store is full")
         self.object_store_id = dataset.object_store_id  # these will be the same thing after the first output
+
+
+def persist_extra_files(object_store: ObjectStore, src_extra_files_path: str, primary_data: "DatasetInstance"):
+    if src_extra_files_path and os.path.exists(src_extra_files_path):
```

Comment:
I don't think `src_extra_files_path` can be an empty string.

```suggestion
    if os.path.exists(src_extra_files_path):
```

Reply:
It certainly could be in the past if the extra files dir hadn't been created yet. Not sure I'm ready remove the safe-guard ?